compiler/ir/dart/access_pattern.py

```diff
@@ -196,6 +196,8 @@ def matches(self, sp: SchedulePattern):
         Check if a given schedule pattern matches this
         template pattern.
         """
+        if sp.num_dims > self.num_dims:
+            sp = sp.inner_dims(self.num_dims)
         if sp.num_dims != self.num_dims:
```

Comment:
maybe make this more explicit?
```suggestion
        elif sp.num_dims < self.num_dims:
```